src/types.ts

```diff
@@ -0,0 +1,49 @@
+import { DocumentNode } from 'graphql/language/ast';
+import * as Observable from 'zen-observable';
+
+export interface GraphQLRequest {
+  //This string could be an id for a persisted query
+  query?: string | DocumentNode;
+  variables?: object;
+  context?: object;
+}
+
+export interface Link {
+  request (operation: Operation, forward?: NextLink): Observable<FetchResult>;
+}
+
+export interface PromiseLink {
+  request (operation: Operation): Promise<FetchResult>;
+}
+
+export interface Operation {
+  query: DocumentNode;
+  variables?: {
+    [variables: string]: any;
+  };
+  operationName?: string;
+  context?: {
+    [meta: string]: any;
+  };
+}
+
+export interface Subscriber<T> {
+  next?: (result: T) => void;
+  error?: (error: any) => void;
+  complete?: () => void;
+}
+
+export interface FetchResult {
+  data: any;
+  errors?: any;
+  extensions?: any;
+  context?: object;
+}
+
+export interface Subscription {
```

Comment:
Lets check to see if `@types/zen-observable` has these so we can use the types of the underlying observable implementation